From 0908bca367e0c6f1f9a0f01788450024b4574aac Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Mon, 26 Dec 2011 23:20:31 +0800
Subject: ENGR00170938-1 mxc hdmi: Enable HDMI output color space convert

Fix HDMI AVI info frame config register bit define error.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 arch/arm/plat-mxc/include/mach/mxc_hdmi.h |   34 ++++++++++++++---------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/arch/arm/plat-mxc/include/mach/mxc_hdmi.h b/arch/arm/plat-mxc/include/mach/mxc_hdmi.h
index 56729b2..02a413f 100644
--- a/arch/arm/plat-mxc/include/mach/mxc_hdmi.h
+++ b/arch/arm/plat-mxc/include/mach/mxc_hdmi.h
@@ -787,26 +787,26 @@ enum {
 	HDMI_FC_PRCONF_OUTPUT_PR_FACTOR_MASK = 0x0F,
 	HDMI_FC_PRCONF_OUTPUT_PR_FACTOR_OFFSET = 0,
 
-/* FC_AVICONF0-FC_AVICONF2 field values */
-	HDMI_FC_AVICONF0_PIX_FMT_MASK = 0x60,
+/* FC_AVICONF0-FC_AVICONF3 field values */
+	HDMI_FC_AVICONF0_PIX_FMT_MASK = 0x03,
 	HDMI_FC_AVICONF0_PIX_FMT_RGB = 0x00,
-	HDMI_FC_AVICONF0_PIX_FMT_YCBCR422 = 0x20,
-	HDMI_FC_AVICONF0_PIX_FMT_YCBCR444 = 0x40,
-	HDMI_FC_AVICONF0_ACTIVE_FMT_MASK = 0x10,
-	HDMI_FC_AVICONF0_ACTIVE_FMT_INFO_PRESENT = 0x10,
+	HDMI_FC_AVICONF0_PIX_FMT_YCBCR422 = 0x01,
+	HDMI_FC_AVICONF0_PIX_FMT_YCBCR444 = 0x02,
+	HDMI_FC_AVICONF0_ACTIVE_FMT_MASK = 0x40,
+	HDMI_FC_AVICONF0_ACTIVE_FMT_INFO_PRESENT = 0x40,
 	HDMI_FC_AVICONF0_ACTIVE_FMT_NO_INFO = 0x00,
 	HDMI_FC_AVICONF0_BAR_DATA_MASK = 0x0C,
 	HDMI_FC_AVICONF0_BAR_DATA_NO_DATA = 0x00,
-	HDMI_FC_AVICONF0_BAR_DATA_VERT_BAR = 0x40,
-	HDMI_FC_AVICONF0_BAR_DATA_HORIZ_BAR = 0x80,
-	HDMI_FC_AVICONF0_BAR_DATA_VERT_HORIZ_BAR = 0xC0,
-	HDMI_FC_AVICONF0_SCAN_INFO_MASK = 0x03,
-	HDMI_FC_AVICONF0_SCAN_INFO_OVERSCAN = 0x01,
-	HDMI_FC_AVICONF0_SCAN_INFO_UNDERSCAN = 0x02,
+	HDMI_FC_AVICONF0_BAR_DATA_VERT_BAR = 0x04,
+	HDMI_FC_AVICONF0_BAR_DATA_HORIZ_BAR = 0x08,
+	HDMI_FC_AVICONF0_BAR_DATA_VERT_HORIZ_BAR = 0x0C,
+	HDMI_FC_AVICONF0_SCAN_INFO_MASK = 0x30,
+	HDMI_FC_AVICONF0_SCAN_INFO_OVERSCAN = 0x10,
+	HDMI_FC_AVICONF0_SCAN_INFO_UNDERSCAN = 0x20,
 	HDMI_FC_AVICONF0_SCAN_INFO_NODATA = 0x00,
 
 	HDMI_FC_AVICONF1_ACTIVE_ASPECT_RATIO_MASK = 0x0F,
-	HDMI_FC_AVICONF1_ACTIVE_ASPECT_RATIO_USE_CODED = 0x09,
+	HDMI_FC_AVICONF1_ACTIVE_ASPECT_RATIO_USE_CODED = 0x08,
 	HDMI_FC_AVICONF1_ACTIVE_ASPECT_RATIO_4_3 = 0x09,
 	HDMI_FC_AVICONF1_ACTIVE_ASPECT_RATIO_16_9 = 0x0A,
 	HDMI_FC_AVICONF1_ACTIVE_ASPECT_RATIO_14_9 = 0x0B,
@@ -840,10 +840,10 @@ enum {
 	HDMI_FC_AVICONF2_IT_CONTENT_VALID = 0x80,
 
 	HDMI_FC_AVICONF3_IT_CONTENT_TYPE_MASK = 0x03,
-	HDMI_FC_AVICONF2_IT_CONTENT_TYPE_GRAPHICS = 0x00,
-	HDMI_FC_AVICONF2_IT_CONTENT_TYPE_PHOTO = 0x01,
-	HDMI_FC_AVICONF2_IT_CONTENT_TYPE_CINEMA = 0x02,
-	HDMI_FC_AVICONF2_IT_CONTENT_TYPE_GAME = 0x03,
+	HDMI_FC_AVICONF3_IT_CONTENT_TYPE_GRAPHICS = 0x00,
+	HDMI_FC_AVICONF3_IT_CONTENT_TYPE_PHOTO = 0x01,
+	HDMI_FC_AVICONF3_IT_CONTENT_TYPE_CINEMA = 0x02,
+	HDMI_FC_AVICONF3_IT_CONTENT_TYPE_GAME = 0x03,
 	HDMI_FC_AVICONF3_QUANT_RANGE_MASK = 0x0C,
 	HDMI_FC_AVICONF3_QUANT_RANGE_LIMITED = 0x00,
 	HDMI_FC_AVICONF3_QUANT_RANGE_FULL = 0x04,
-- 
1.7.9.5

